components/script/dom/notification.rs

```diff
@@ -117,11 +117,30 @@ impl Notification {
         let body = options.body.clone();
         let tag = options.tag.clone();
 
-        // TODO: image, icon, badge should call `ServoUrl::parse_with_base` with
-        //       baseURL extract from `environment settings object`
-        let image = None;
-        let icon = None;
-        let badge = None;
+        // If options["image"] exists, then parse it using baseURL, and if that does not return failure,
+        // set notification’s image URL to the return value. (Otherwise notification’s image URL is not set.)
+        let image = options.image.as_ref().and_then(|image_url| {
+            // TODO: call `ServoUrl::parse_with_base` with baseURL extracted from `environment settings object`
+            ServoUrl::parse(image_url.as_ref())
+                .map(|url| USVString::from(url.to_string()))
+                .ok()
+        });
+        // If options["icon"] exists, then parse it using baseURL, and if that does not return failure,
+        // set notification’s icon URL to the return value. (Otherwise notification’s icon URL is not set.)
+        let icon = options.icon.as_ref().and_then(|icon_url| {
+            // TODO: call `ServoUrl::parse_with_base` with baseURL extracted from `environment settings object`
```

Comment:
https://github.com/servo/servo/blob/ffccc5c88b8d47efc777b0bc9d2620bed4bc84d7/components/script/dom/globalscope.rs#L2348

Reply:
Thank you!

I have a question that should we assume the `environment settings object` here will always be `globalscope`?

Reply:
I believe so!